plugins/modules/zypper.py

```diff
@@ -349,15 +370,42 @@ def parse_zypper_xml(m, cmd, fail_not_found=True, packages=None):
             # run zypper again with the same command to complete update
             return parse_zypper_xml(m, cmd, fail_not_found=fail_not_found, packages=packages)
 
+        # apply simple_errors logic to rc 0,102,103,106
+        if m.params['simple_errors']:
+            simple_errors = get_simple_errors(dom)
+            if simple_errors is not None:
+                stdout = simple_errors
```

Comment:
This can be simplified to
```suggestion
            stdout = get_simple_errors(dom) or stdout
```
but if you prefer the longer form (which is easier to understand for folks who aren't that familiar with using `None` in boolean expressions) that's also fine.